release-25.4: changefeedccl: disable DB-level feeds

KeithCh opened the pull request:
DB-level feeds is not ready to be released to the public yet. We will 
backport this PR to 25.4 to prevent this feature from being used.

Resolves: #152998
Epic: CRDB-1421

Release note: None
Release justification: Prevent use of unreleased feature.

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/154059)
<!-- Reviewable:end -->

KeithCh committed:
changefeedccl: disable DB-level feeds

DB-level feeds is not ready to be released to the public yet. We will
backport this PR to 25.4 to prevent this feature from being used.

Resolves: #152998
Epic: CRDB-1421

Release note: none

andyyang890 commented on pkg/ccl/changefeedccl/changefeed_test.go:
```diff
@@ -168,59 +168,10 @@ func TestDatabaseLevelChangefeedBasics(t *testing.T) {
-			`foo2: [1]->{"after": null}`,
-		})
+		expectErrCreatingFeed(t, f,
+			`CREATE CHANGEFEED FOR DATABASE d EXCLUDE TABLES foo`,
```
nit: the original test doesn't test EXCLUDE TABLES
```suggestion
			`CREATE CHANGEFEED FOR DATABASE d`,
```

andyyang890 commented on pkg/ccl/changefeedccl/changefeed_test.go:
```diff
@@ -1165,15 +1025,20 @@ func TestChangefeedMissingDatabaseErr(t *testing.T) {
 	cdcTest(t, func(t *testing.T, s TestServer, f cdctest.TestFeedFactory) {
-		expectErrCreatingFeed(t, f, `CREATE CHANGEFEED FOR DATABASE system`, `changefeed cannot target the system database`)
+		expectErrCreatingFeed(t, f,
+			`CREATE CHANGEFEED FOR DATABASE d`,
```
nit: in keeping with the spirit of the original test, the changefeed should target the system database
```suggestion
			`CREATE CHANGEFEED FOR DATABASE system`,
```

KeithCh commented:
bors r=andyyang890,asg0451
